File: 2024 December/Sorted and Rotated Minimum/main.cpp
// Link: https://www.geeksforgeeks.org/problems/minimum-element-in-a-sorted-and-rotated-array3611/1

#include <iostream>
#include <vector>
#include <climits>
using namespace std;

class Solution
{
public:
    int findMin(vector<int> &arr)
    {
        int mini = INT_MAX;
        for (int i = 0; i < arr.size(); i++)
        {
            mini = min(mini, arr[i]);
        }
        return mini;
    }
};

int main()
{
    Solution object;
    vector<int> arr = {5, 6, 1, 2, 3, 4};
    cout << object.findMin(arr) << endl;
    arr = {3, 2, 2, 2};
    cout << object.findMin(arr) << endl;
    arr = {4, 4, 4};
    cout << object.findMin(arr) << endl;
    return 0;
}

File: 2024 December/Sorted and Rotated Minimum/main.py
# Link: https://www.geeksforgeeks.org/problems/minimum-element-in-a-sorted-and-rotated-array3611/1


class Solution:
    def findMin(self, arr):
        mini = float("inf")
        for num in arr:
            mini = min(mini, num)
        return mini


if __name__ == "__main__":
    object = Solution()
    arr = [5, 6, 1, 2, 3, 4]
    print(object.findMin(arr))
    arr = [3, 2, 2, 2]
    print(object.findMin(arr))
    arr = [4, 4, 4]
    print(object.findMin(arr))
